### PR TITLE
By-pass set_annotations when plotting ICA sources

### LINKED ISSUE #11762: Annotation displayed on ICA sources when meas_date is None and first_samp is not 0 are wrong
### Description of the problem

One more time-management related issue with annotations. MWE:

```
from datetime import datetime, timezone

import numpy as np
from mne import Annotations, create_info
from mne.io import RawArray
from mne.preprocessing import ICA


#%%
# date = datetime(1995, 3, 28, 0, 0, 0, 0, tzinfo=timezone.utc)

info = create_info(5, 1000, "eeg")
data = np.random.randn(5, 10000)
raw = RawArray(data, info)
assert raw.first_samp == 0
# raw.set_meas_date(date)
annotation = Annotations(onset=4, duration=1, description="bad_test")
raw.set_annotations(annotation)
# raw.plot(theme="light", scalings="auto")

#%%
raw.crop(2, 8)
# raw.plot(theme="light", scalings="auto")

#%%
ica = ICA(
    method="picard", fit_params=dict(ortho=False, extended=True), random_state=888
)
ica.fit(raw, picks="eeg", reject_by_annotation=True)
ica.plot_sources(raw, theme="light")
```

If there is no measurement date and if `raw.first_samp` is different from `0` (e.g. by cropping), the annotations displayed in `ica.plot_sources` are not correctly set. Issue is here: https://github.com/mne-tools/mne-python/blob/7def8369d7ca691d807fe916663168c2ee4c10e0/mne/viz/ica.py#L1304-L1305

Still thinking about what is a good fix.. time-management is a real headache :exploding_head:

### PR DESCRIPTION
**Pull request by mscheltienne**
Fixes #11762
Tested in the following scenario:
- `raw.first_samp` set to 0, with and without measurement date
- `raw.first_samp` different from 0, with and without measurement date

I'll try to add some tests when able.

### CODE REVIEW

**Comment by mscheltienne** [comment]
That should do it, test are failing or main and passing in this PR because in `ica.plot_sources(raw)` the annotation ends up being outside of the `raw` range (raw of 1s, annotation shifted to 43.1s due to `first_samp` and lack of `meas_date`). 
So we don't even enter in the part which checks that the region location is indeed `(0.2, 0.3)`.. but that's a bonus :smile:

File: mne/viz/tests/test_ica.py [hunk 1]
```diff
@@ -328,6 +328,22 @@ def test_plot_ica_sources(raw_orig, browser_backend, monkeypatch):
...
+    raw.set_meas_date(None)
+    assert raw.first_samp != 0
+    raw.set_annotations(Annotations([0.2], [0.1], "Test"))
+    fig = ica.plot_sources(raw)
+    if browser_backend.name == "matplotlib":
+        assert len(fig.mne.ax_main.collections) == 1
+        assert len(fig.mne.ax_hscroll.collections) == 1
+    else:
+        assert len(fig.mne.regions) == 1
+        assert np.allclose(fig.mne.regions[0].getRegion(), (0.2, 0.3))
```
Comment (larsoner) [inline mne/viz/tests/test_ica.py:344]: Do not use `assert np.allclose` in tests but rather `assert_allclose` -- it has a sane default `atol=0` and also more informative message on failure
```suggestion
        assert_allclose(fig.mne.regions[0].getRegion(), (0.2, 0.3))
```

File: mne/viz/tests/test_ica.py [hunk 2]
```diff
@@ -328,6 +328,22 @@ def test_plot_ica_sources(raw_orig, browser_backend, monkeypatch):
         assert len(fig.mne.ax_hscroll.collections) == 1
     else:
         assert len(fig.mne.regions) == 1
+        assert np.allclose(fig.mne.regions[0].getRegion(), (0.2, 0.3))
```
Comment (larsoner) [inline mne/viz/tests/test_ica.py:331]: Same here. Then you'll need to import at the top.

I tried to push the change but GitHub wouldn't let me for some reason

File: mne/viz/tests/test_ica.py (hunk 1 above)
Comment (mscheltienne) [reply mne/viz/tests/test_ica.py:344]: Thanks for the tip, I'll look into `numpy.testing`, looks useful :)

File: mne/viz/tests/test_ica.py (hunk 2 above)
Comment (mscheltienne) [reply mne/viz/tests/test_ica.py:331]: I can't either.. well that's interesting.

**Pull request closed without merging**